pkg/gotohelm/helmette/sprig_test.go

```diff
@@ -34,3 +39,156 @@ func TestDefault(t *testing.T) {
+			helmette.Merge(
+				map[string]any{"label": "something-default"},
+				map[string]any{"label": "overwritten"},
+				map[string]any{"label": "last-overwritten"}))
+	})
+
+	t.Run("empty_map_as_destination", func(t *testing.T) {
+		dest := make(map[string]string)
+		helmette.Merge(
+			dest,
+			map[string]string{"label": "first-in-line"},
+			map[string]string{"label": "last-value-for-the-label-key"})
+		require.Equal(t, map[string]string{"label": "first-in-line"}, dest)
+	})
+
+	t.Run("multiple_keys", func(t *testing.T) {
+		dest := make(map[string]string)
+		helmette.Merge(
+			dest,
+			map[string]string{"label": "first-in-line", "a": "test"},
+			map[string]string{"label": "last-value-for-the-label-key", "b": "test"})
+		require.Equal(t, map[string]string{"label": "first-in-line", "a": "test", "b": "test"}, dest)
+	})
+
+	t.Run("none_empty_map", func(t *testing.T) {
+		noneEmpty := map[string]any{"key": "value"}
+		helmette.Merge(
+			noneEmpty,
+			map[string]any{"label": "overwritten"},
+			map[string]any{"label": "last-overwritten"})
+		require.Equal(t, map[string]any{"label": "overwritten", "key": "value"}, noneEmpty)
+	})
+
+	t.Run("copy_of_sprig_merge_test_case", func(t *testing.T) {
+		dict := map[string]any{
+			"src2": map[string]any{
+				"h": 10,
+				"i": "i",
+				"j": "j",
+			},
+			"src1": map[string]any{
+				"a": 1,
+				"b": 2,
+				"d": map[string]any{
+					"e": "four",
+				},
+				"g": []int{6, 7},
+				"i": "aye",
+				"j": "jay",
+				"k": map[string]any{
+					"l": false,
+				},
+			},
+			"dst": map[string]any{
+				"a": "one",
+				"c": 3,
+				"d": map[string]any{
+					"f": 5,
+				},
+				"g": []int{8, 9},
+				"i": "eye",
+				"k": map[string]any{
+					"l": true,
+				},
+			},
+		}
+		tpl := `{{merge .dst .src1 .src2 | toRawJson }}`
+		output, err := runRawTemplate(tpl, dict)
+		require.NoError(t, err)
+
+		ts := make(map[string]any)
+
+		err = json.Unmarshal([]byte(output), &ts)
+		require.NoError(t, err)
+
+		expected := map[string]any{
+			"a": "one", // key overridden
+			"b": 2,     // merged from src1
+			"c": 3,     // merged from dst
+			"d": map[string]any{ // deep merge
+				"e": "four",
+				"f": 5,
+			},
+			"g": []int{8, 9}, // overridden - arrays are not merged
+			"h": 10,          // merged from src2
+			"i": "eye",       // overridden twice
+			"j": "jay",       // overridden and merged
+			"k": map[string]any{
+				"l": true, // overridden
+			},
+		}
+
+		answer := helmette.Merge(map[string]any{
+			"a": "one",
+			"c": 3,
+			"d": map[string]any{
+				"f": 5,
+			},
+			"g": []int{8, 9},
+			"i": "eye",
+			"k": map[string]any{
+				"l": true,
+			},
+		}, dict["src1"].(map[string]any), dict["src2"].(map[string]any))
+
+		assert.Equal(t, changeAnyIntToFloat64(expected), ts)
+		assert.Equal(t, expected, dict["dst"])
+		assert.Equal(t, expected, answer)
+	})
+}
+
+func changeAnyIntToFloat64(input map[string]any) map[string]any {
```

Comment:
Same nit as the other PR: can this instead use `json.Marshal/json.Unmarshal`? It might be worth making a `JSONRoundtrip(any) map[string]any` function as I suspect we'll be doing this a lot.

Reply:
Sorry, I cherry-pick from other commit and forgot to address that! I agree with you and I will push the change shortly